runtime/gc_structs/ClassLoaderClassesIterator.cpp

```diff
@@ -108,13 +152,18 @@ GC_ClassLoaderClassesIterator::nextClass()
 		if (ANONYMOUS_CLASSES == _mode) {
 			_nextClass = nextAnonymousClass();
 		} else {
-			if ( (result->classLoader == _classLoader) && (NULL != result->arrayClass) ) {
+			J9Class* array = nextArrayClass();
+			if ((result->classLoader == _classLoader) && (NULL != array)) {
 				/* this class is defined in the loader, so follow its array classes */
-				_nextClass = result->arrayClass;
+				_nextClass = array;
 			} else if (TABLE_CLASSES == _mode) {
 				_nextClass = nextTableClass();
+				_startingClass = _nextClass;
+				_arrayState = 0;
```

Comment:
classLoaderClassIterator_state_valueTypeArray should be used instead of 0

Reply:
BTW these states are private, so can use names without `classLoaderClassIterator` prefix if you like